elytron/ELY-1686-legacy-properties-files-into-FileSystemRealm-utility.adoc

```diff
@@ -0,0 +1,174 @@
+--output-location/-o <outDirectory> [--filesystem-realm-name/-f] <name>
+[--security-domain-name/-s] <name> [--http-authentication-factory-name/-h] <name>
+[--application-security-domain-name/-a] <name>
+
+* There should be a text descriptor file that allows multiple inputs to be specified. The
+descriptor file should only accept the long-form option names for readability. Using the
+text descriptor file will have its own option, --bulk-convert/-b
+
+** CLI Command
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --bulk-convert/-b <file>
+
+** Text descriptor file
+[source,text]
+----
+user-file:/relative/path/to/userFile1
+roles-file:/relative/path/to/rolesFile1
+output-location:/relative/path/to/output/outDirectory1
+filesystem-realm-name:nameOfFileSystemRealm1
+security-domain-name:nameOfSecurityDomain1
+http-authentication-factory-name:nameOfHttpAuthenticationFactory1
+application-security-domain-name:nameOfApplicationSecurityDomain1
+
+user-file:/absolute/path/to/userFile2
+roles-file:/absolute/path/to/rolesFile2
+output-location:/absolute/path/to/output/outDirectory2
+filesystem-realm-name:nameOfFileSystemRealm2
+security-domain-name:nameOfSecurityDomain2
+http-authentication-factory-name:nameOfHttpAuthenticationFactory2
+application-security-domain-name:nameOfApplicationSecurityDomain2
+----
+
+* Summary of options and their description
+|===
+|Long-form |Short-form |Description
+
+| --user-file
+| -u
+| The relative or absolute path to the users file
+
+| --roles-file
+| -r
+| The relative or absolute path to the roles file
+
+| --output-location
+| -o
+| The relative or absolute path to the output directory
+
+| --filesystem-realm-name
+| -f
+| Name of the filesystem-realm to be configured
+
+| --security-domain-name
+| -s
+| Name of the security-domain to be configured
+
+| --http-authentication-factory-name
+| -h
+| Name of the http-authentication-factory to be configured
+
+| --application-security-domain-name
+| -a
+| Name of the application-security-domain to be configured
+
+| --bulk-convert
+| -b
+| Tells the command to use a descriptor file, for multiple users-roles file combinations
+can be provided.
+|===
+
+=== Nice-to-Have Requirements
+
+* The command should have a silent option, similar to the -s option in the unix make
+command, where only errors and prompts are printed.
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm [--silent] --bulk-convert/-b <file>
+
+* The command should have a summary option that prints out detailed information on
+every operation that was performed, once the command has finished converting all files.
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm [--summary] --bulk-convert/-b <file>
+
+* Summary of nice-to-have options and their description
+|===
+|Long-form |Short-form |Description
+
+| --silent
+| N/A
+| Suppresses all output except errors and prompts
+
+| --summary
+| N/A
+| Provides a detailed summary of all operations performed, once the command finishes.
```

Comment:
Are --silent and --summary mutually exclusive?

Reply:
I think it would make sense to have them mutually exclusive, but I see no reason they have to be. --silent would suppress the runtime output, while --summary would give an output of everything that happened, after the command finishes converting all files.

Reply:
Yeah both ways make sense here, hence my question. Do we have any precedent from other similar commands in WF or other JBoss projects?

Reply:
WF has a silent option and some commands in WildFly-Elytron-Tool have a summary option, but I'm not aware of anywhere that they're both available.